src/groups/mqb/mqbblp/mqbblp_clusterqueuehelper.cpp

```diff
@@ -178,6 +160,40 @@ void countUnconfirmed(bsls::Types::Int64* result, mqbi::Queue* queue)
 
 }  // close unnamed namespace
 
+// -------------------------------------------
+// struct ClusterQueueHelper::OpenQueueContext
+// -------------------------------------------
+
+ClusterQueueHelper::OpenQueueContext::OpenQueueContext(
+    mqbi::Domain*                                             domain,
+    const bmqp_ctrlmsg::QueueHandleParameters&                handleParameters,
+    const bsl::shared_ptr<mqbi::QueueHandleRequesterContext>& clientContext,
+    const mqbi::Cluster::OpenQueueCallback&                   callback)
+: d_queueContext_p(0)
+, d_domain_p(domain)
+, d_handleParameters(handleParameters)
+, d_upstreamSubQueueId(bmqp::QueueId::k_UNASSIGNED_SUBQUEUE_ID)
+, d_clientContext(clientContext)
+, d_callback(callback)
+{
+    // NOTHING
+}
+
+ClusterQueueHelper::OpenQueueContext::~OpenQueueContext()
+{
+    BSLS_ASSERT_SAFE(d_queueContext_p);
+
+    --d_queueContext_p->d_liveQInfo.d_inFlight;
```

Comment:
This class API still allows to copy or assign the object to another one, and it might break refcounting done in destructor. I think copy/assignment constructors should be deleted to prevent this. It will also safeguard against non-obvious problems with temporal object copies